src/Client/Updater.php

```diff
@@ -112,13 +102,81 @@ public function validateTarget($targetRepoPath, $targetStream)
         $timestampStructure = json_decode($timestampContents, true);
         // SPEC: 2.1
         if (! $this->checkSignatures($timestampStructure, 'timestamp')) {
-          // Exception? Log + return false?
             throw new \Exception("Improperly signed repository timestamp.");
         }
 
+        // SPEC: 2.2
+        $currentStateTimestamp = json_decode($durableStorage['timestamp.json'], true);
+        $this->checkRollbackAttack($currentStateTimestamp['signed'], $timestampStructure['signed']);
+
+        // SPEC: 2.3
+        $this->checkFreezeAttack($timestampStructure['signed'], $nowDate);
+        $durableStorage['timestamp.json'] = $timestampContents;
+
         return true;
     }
 
+    protected function metadataTimestampToDatetime(string $timestamp) : \DateTimeImmutable
+    {
+        $dt = \DateTimeImmutable::createFromFormat("Y-m-d\TH:i:sT", $timestamp);
+        if ($dt === false) {
+            throw new FormatException($timestamp, "Could not be interpreted as a DateTime");
+        }
+        return $dt;
+    }
+
+    /**
+     * Verifies that an incoming remote version of a metadata file is >= the last known version.
+     *
+     * @param $localMetadata
+     * @param $remoteMetadata
+     * @throws RollbackAttackException
+     */
+    protected function checkRollbackAttack(array $localMetadata, array $remoteMetadata)
+    {
+        $localVersion = $localMetadata['version'] + 0;
```

Comment:
I agree with Ted that this is a rather opaque line. If the intent is to cast it to an int, why not just `(int)`?

Reply:
Muscle memory when casting strings to numeric in PHP, I usually use this way because it universally works for ints and floats. TUF constrains us to ints here so yeah we can use `(int)`